packages/@uppy/tus/src/index.ts

```diff
@@ -357,7 +442,7 @@ export default class Tus extends BasePlugin {
 
       uploadOptions.metadata = meta
 
-      upload = new tus.Upload(file.data, uploadOptions)
+      upload = new tus.Upload(file.data, uploadOptions as tus.UploadOptions)
```

Comment:
```suggestion
      upload = new tus.Upload(file.data, uploadOptions as tus.UploadOptions) // todo type uploadOptions properly instead
```

Reply:
That's not possible here, as we change `onShouldRetry` and `onBeforeRequest` in a non breaking way and pass it to tus. That means the types don't match but it's intentional and we have to cast something somewhere.

Reply:
the types are not the same.
- uppy's `onShouldRetry: ((err: tus.DetailedError, retryAttempt: number, options: TusOpts<M, B>, next: (e: tus.DetailedError) => void) => boolean) | undefined`
- tus's `onShouldRetry?: ((error: tus.DetailedError, retryAttempt: number, options: tus.UploadOptions) => boolean) | null | undefined`

there is a difference in `options` as well as the extra argument `next` in uppy. if we pass this `next` argument to tus (because we disable typechecking) that might work fine now, but if tus in the future adds a 4th argument to this function (that is not a `next` function) we will probably end up without any typescript error but it will crash at runtime.